CIPs/0010.md

```diff
@@ -0,0 +1,56 @@
+# CIP 0010: General Account Signer Authorization
+
+- Date: 2020-05-13
+- Author: @nambrot
+- Status: WIP
+
+## Overview
+
+The `Accounts.sol` smart contract houses a general account abstraction that currently services the following purposes:
+- Acts as the canonical account for the Proof-of-stake system (aka LockedGold account)
+- keeps track of authorized signer keys in the PoS system (validator, voter and attestation signers)
+- stores basic profile information such as a name and mapping to `dataEncryptionKey` and `walletAddress`
+- entry into an account's off-chain metadata
+
+Over the past months, it has become noticable that especially the authorization of signer keys by the "account key" is a very useful method to authenticate the "account owner" without having to unncessarily bring the "account key" online which is the most priveledged, and thus most important to secure. Example of such cases have been:
+
+- The signing of the off-chain metadata should ideally not happen with the account key. During the development of the `RelaeseGold` smart contract it has been found that smart contracts as accounts cannot sign metadata. As a workaround, verifying metadata has been adjusted to allow any authorized signer to sign the payload, which violates the intention behind having well scoped authrized signers in the first place.
+- As part of PGPNP, accounts have to authenticate with the PGPNP service for rate-limiting purposes. To do that, the "account key" has to sign the request which also violates the intention of keeping keys that can move funds as cold as possible.
+- For other applications such as Profile Picture/Names or more down-the-road applications such as messaging, specific encryption keys are going to be useful. Additionally, the `dataEncryptionKey` seems out of place, despite it being effectively also an authorized key (if not signer).
```

Comment:
Agree it's out of place but note that it's a public key and not an address. How does this CIP address that?

Reply:
This is not explicitly specified here, but the address of the `dataEncryptionKey` can be just another signer of the account?